config-model/src/test/java/com/yahoo/schema/processing/RankingExpressionTypeResolverTestCase.java

```diff
@@ -458,6 +458,54 @@ void undeclaredQueryFeaturesAreAcceptedWithWarningWhenUsingTensors() throws Exce
                      message);
     }
 
+    @Test
+    void requireThatUsingArrayWarns() throws Exception {
+        InspectableDeployLogger logger = new InspectableDeployLogger();
+        ApplicationBuilder builder = new ApplicationBuilder(logger);
+        builder.addSchema(joinLines(
+                "search test {",
+                "  document test { ",
+                "    field foo type array<float> {",
+                "      indexing: attribute",
+                "    }",
+                "  }",
+                "  rank-profile my_rank_profile {",
+                "    first-phase {",
+                "      expression: map(attribute(foo), f(x)(42*x))",
+                "    }",
+                "  }",
+                "}"
+        ));
+        builder.build(true);
+        String message = logger.findMessage("collection");
+        assertNotNull(message);
+        assertEquals("WARNING: Using attribute(foo) collectiontype: ARRAY as ranking expression input", message);
+    }
+
+    @Test
+    void requireThatUsingWsetWarns() throws Exception {
+        InspectableDeployLogger logger = new InspectableDeployLogger();
+        ApplicationBuilder builder = new ApplicationBuilder(logger);
+        builder.addSchema(joinLines(
+                "search test {",
+                "  document test { ",
+                "    field foo type weightedset<int> {",
+                "      indexing: attribute",
+                "    }",
+                "  }",
+                "  rank-profile my_rank_profile {",
+                "    first-phase {",
+                "      expression: attribute(foo)",
```

Comment:
What will happen if you use `attribute(foo).count`? This is a supported feature for both array and weighted set. From https://docs.vespa.ai/en/reference/rank-features.html#document-features:

_attribute(name).count - The number of elements in the attribute with the given name_

Reply:
Discussed with Arne, and this is handled.